src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -243,6 +239,69 @@ kind switch
                     originalText[closeQuoteRange],
                     originalToken.GetTrailingTrivia());
             }
+
+            static SyntaxDiagnosticInfo makeIndentationError(
+                ReadOnlySpan<char> indentationWhitespace,
+                ReadOnlySpan<char> currentLineWhitespace,
+                int lineStartIndex,
+                int currentIndex)
+            {
+                // Specialized error message if this is a spacing difference.
+                if (CheckForSpaceDifference(
+                        currentLineWhitespace, indentationWhitespace,
+                        out var currentLineWhitespaceChar, out var indentationWhitespaceChar))
+                {
+                    return MakeError(
+                        lineStartIndex,
+                        width: currentIndex - lineStartIndex,
+                        ErrorCode.ERR_LineContainsDifferentWhitespace,
+                        currentLineWhitespaceChar, indentationWhitespaceChar);
+                }
+                else
+                {
+                    return MakeError(
+                        lineStartIndex,
+                        width: currentIndex - lineStartIndex,
+                        ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
+                }
+            }
+
+            // if the interpolation starts on its own line, then it has to have correct indentation whitespace
+            // before it.  e.g.:
+            //
+            //      var x = """
+            //          {1 + 1}
+            //          """
+            //
+            // Not:
+            //
+            //      var x = """
+            // {1 + 1}
+            //          """
+            //
+            // Note: We don't need to check
```

Comment:
The example that we don't need to check is exactly the same as the example just above, which we are checking? I'm not sure how I'm supposed to read this comment.

Reply:
the example below has leading whitespace.  i'll doc it to make it clearer.

Reply:
doc'ed